File: java/src/main/java/deltix/luminary/implementation/InterfacePropertyNode.java
package deltix.luminary.implementation;

import java.util.List;

public class InterfacePropertyNode extends PropertyNode {
    private final boolean isOverride;
    private final boolean isGettable;
    private final boolean isSettable;

    public InterfacePropertyNode(String name, String type, boolean isOverride, boolean isGettable, boolean isSettable,
                                 List<String> comments) {
        super(name, type, comments);
        this.isOverride = isOverride;
        this.isGettable = isGettable;
        this.isSettable = isSettable;
    }

    public boolean isOverride() {
        return isOverride;
    }

    public boolean isGettable() {
        return isGettable;
    }

    public boolean isSettable() {
        return isSettable;
    }
}

File: csharp/main/Implementation/InterfacePropertyNode.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Deltix.Luminary.Implementation
{
	internal class InterfacePropertyNode : PropertyNode
	{
		public InterfacePropertyNode([NotNull] String name, [NotNull] String type, Boolean isOverride, Boolean isGettable, Boolean isSettable,
			[CanBeNull] List<String> comments)
			: base(name, type, comments)
		{
			IsOverride = isOverride;
			IsGettable = isGettable;
			IsSettable = isSettable;
		}

		public Boolean IsOverride { get; }

		public Boolean IsGettable { get; }

		public Boolean IsSettable { get; }
	}
}
